https://stepik.org/lesson/573098/step/8

Итак значение величины $F$ одинаково для обоих случаев:

1. `precision=0.7` `recall=0.8`

2. `precision=0.8` `recall=0.7`

В задачах, где precision и recall имеют разную значимость (например в задаче о выявлении больных: признать больного здоровым - это более серьезная ошибка, чем признать больным здорового) **применяют характеристику** $F_β$.

Вот и мы также поступим. Найдите для $β=2$ значение $F_{\beta}$ для `precision=0`.7 `recall=0.8`, а потом для случая `precision=0.8` `recall=0.7`.

У вас получится в итоге два числа. В ответ напишите максимальное из них.

In [ ]:
# 0.78
# max((1 + 2**2) * 0.7 * 0.8 / (2**2 * 0.7 + 0.8), (1 + 2**2) * 0.8 * 0.7 / (2**2 * 0.8 + 0.7))
max((1 + beta**2) * precision * recall / (beta**2 * precision + recall) \
    for beta, precision, recall in [[2, .7, .8], [2, .8, .7]])

0.7777777777777779

# Решение

$$
F_{\beta} = \frac {(1 + \beta^2) \cdot precision \cdot recall}{\beta^2 \cdot precision + recall}
$$

$$~$$

$$
\frac {(1 + 2^2) \cdot 0.7 \cdot 0.8}{2^2 \cdot 0.7 + 0.8} = 0.7778
$$

$$~$$

$$
\frac {(1 + 2^2) \cdot 0.8 \cdot 0.7}{2^2 \cdot 0.8 + 0.7} = 0.7179
$$
